### DISCUSSION TITLE
Interval data type: intervalstyle support

### DISCUSSION THREAD

**Discussion by vslio** [supabase/supabase#17073] [Questions]
Howdy!

I have been trying to figure out a solution to something for a while, but I haven't gotten anywhere so I could use the extra help from the community. 

I am building a React Native app using Supabase, so I'm using `supabase-js`. I have created a column of type `interval` on one of my tables. However, I want to set the [output style of the interval](https://www.postgresql.org/docs/current/datatype-datetime.html#DATATYPE-INTERVAL-OUTPUT) to something other than the default.

This doesn't seem possible at the moment. The only option I see is to maybe create a `View` for my `Select`, and add something like this `SET intervalstyle = 'iso_8601';` right before the select query. 

Any other options? Could I somehow pass specific options like the above (`SET intervalstyle = 'iso_8601';`) when I initialise the client?

I also looked into [luxon](https://moment.github.io/luxon/#/) and other libraries, but they can't parse the default postgres output style.

Thank you for your time. 🙂

**Comment by abeisleem** [comment]
Facing the same issue. Only things I've been able to achieve is using the below snippet to read the intervalstyle. But I can't set it. 
``` sql
select
  current_setting('intervalstyle') as interval_style;
```

**Reply by vslio** [reply]
I ended up writing a small function that converts the postgres interval format to a format I can actually use (in my case for luxon). Not ideal, but I don't see this being a priority for the supabase team.

**Reply by abeisleem** [reply]
Cool. Shortly after replying I ended up finding a [postgres-interval](https://www.npmjs.com/package/postgres-interval) package that did the conversion very nicely. Tested it and it works great.
